src/Microsoft.Management.Configuration/ConfigurationProcessor.cpp

```diff
@@ -626,6 +627,70 @@ namespace winrt::Microsoft::Management::Configuration::implementation
         return *result;
     }
 
+    Configuration::GetAllConfigurationUnitSettingsResult ConfigurationProcessor::GetAllUnitSettings(const ConfigurationUnit& unit)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+        return GetAllUnitSettingsImpl(unit);
+    }
+
+    Windows::Foundation::IAsyncOperation<Configuration::GetAllConfigurationUnitSettingsResult> ConfigurationProcessor::GetAllUnitSettingsAsync(const ConfigurationUnit& unit)
+    {
+        THROW_HR_IF(E_NOT_VALID_STATE, !m_factory);
+
+        auto strong_this{ get_strong() };
+        ConfigurationUnit localUnit = unit;
+
+        co_await winrt::resume_background();
+
+        co_return GetAllUnitSettingsImpl(localUnit, { co_await winrt::get_cancellation_token() });
+    }
+
+    Configuration::GetAllConfigurationUnitSettingsResult ConfigurationProcessor::GetAllUnitSettingsImpl(
+        const ConfigurationUnit& unit,
+        AppInstaller::WinRT::AsyncCancellation cancellation)
+    {
+        auto threadGlobals = m_threadGlobals.SetForCurrentThread();
+
+        IConfigurationSetProcessor setProcessor = m_factory.CreateSetProcessor(nullptr);
+        auto result = make_self<wil::details::module_count_wrapper<implementation::GetAllConfigurationUnitSettingsResult>>();
+        auto unitResult = make_self<wil::details::module_count_wrapper<implementation::ConfigurationUnitResultInformation>>();
+        result->ResultInformation(*unitResult);
+
+        cancellation.ThrowIfCancelled();
+
+        IConfigurationUnitProcessor2 unitProcessor;
+
+        try
+        {
+            // TODO: Directives overlay to prevent running elevated for get
+            unitProcessor = setProcessor.CreateUnitProcessor(unit).as<IConfigurationUnitProcessor2>();

```

Comment:
RE: my other comment on changing the name of the interface.  This would become `try_as` and when `!unitProcessor` would set the consistent error in the event of it not being supported.